### PR TITLE
Async

### PR DESCRIPTION
**Pull request by maldrake**
This PR makes Warpgrapher fully async.

### CODE REVIEW

File: examples/dynamic_rels/main.rs [hunk 1]
```diff
@@ -27,36 +27,36 @@ model:
...
-fn resolve_project_top_contributor(facade: ResolverFacade<()>) -> ExecutionResult {
-    // create dynamic dst node
-    let mut top_contributor_props = HashMap::<String, Value>::new();
-    top_contributor_props.insert(
-        "id".to_string(),
-        Value::from(Uuid::new_v4().to_hyphenated().to_string()),
-    );
-    top_contributor_props.insert("name".to_string(), Value::from("user0".to_string()));
-    let top_contributor = facade.create_node("User", top_contributor_props);
+fn resolve_project_top_contributor(facade: ResolverFacade<'_, ()>) -> BoxFuture<ExecutionResult> {
```
Comment (shellforge) [inline examples/dynamic_rels/main.rs]: @maldrake : why is a lifetime needed in the ResolverFacade here but not in the same place in the `dynamic_props` example?

File: src/engine/schema.rs [hunk 2]
```diff
@@ -2307,6 +2303,11 @@ fn generate_schema(c: &Configuration) -> Result<HashMap<String, NodeType>, Error
         NodeType::new("Query".to_string(), TypeKind::Object, query_props),
     );
 
+    nthm.insert(
```
Comment (shellforge) [inline src/engine/schema.rs:2306]: @maldrake : what is this type for?

File: examples/dynamic_rels/main.rs (hunk 1 above)
Comment (maldrake) [reply examples/dynamic_rels/main.rs]: Good catch. That lifetime marking isn't needed.  I removed it in the latest push.

File: src/engine/schema.rs (hunk 2 above)
Comment (maldrake) [reply src/engine/schema.rs:2306]: The new version of Juniper supports GraphQL subscriptions. The `RootNode` type has been expanded to include a root Query, root Mutation, and root Subscription. It's required for the root node, even if subscriptions are unused, so we include an empty root node with no children.

**Review by shellforge — APPROVED** [review]

**Pull request merged**